File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/QrFormField.java
package com.skedgo.tripkit.booking;

import android.os.Parcel;

import com.google.gson.annotations.SerializedName;

import androidx.annotation.Nullable;

public class QrFormField extends FormField {
    public static final Creator<QrFormField> CREATOR = new Creator<QrFormField>() {
        @Override
        public QrFormField createFromParcel(Parcel in) {
            in.readInt();
            return new QrFormField(in);
        }

        @Override
        public QrFormField[] newArray(int size) {
            return new QrFormField[size];
        }
    };

    @SerializedName("value")
    private String value;

    public QrFormField(Parcel in) {
        super(in);
        this.value = in.readString();
    }

    public QrFormField() {
        super();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(QRCODE);
        super.writeToParcel(dest, flags);
        dest.writeString(value);
    }

    @Nullable
    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

}

File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/QrFormField.kt
package com.skedgo.tripkit.booking

import android.os.Parcel
import android.os.Parcelable.Creator
import com.google.gson.annotations.SerializedName

class QrFormField : FormField {

    @SerializedName("value")
    var mValue: String? = null

    constructor(parcel: Parcel) : super( parcel) {  // Call the parent class constructor
        mValue = parcel.readString()
    }

    // Secondary empty constructor (the default constructor)
    constructor() : super() {
        // Initialize any default values or leave it empty
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(QRCODE)
        super.writeToParcel(dest, flags)
        dest.writeString(mValue)
    }

    override fun getValue(): String? {
        return mValue
    }

    fun setValue(value: String) {
        this.mValue = value
    }

    companion object CREATOR : Creator<QrFormField> {
        override fun createFromParcel(parcel: Parcel): QrFormField {
            parcel.readInt() // Read QRCODE type
            return QrFormField(parcel)
        }

        override fun newArray(size: Int): Array<QrFormField?> {
            return arrayOfNulls(size)
        }
    }
}
